x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Literal.java

```diff
@@ -68,7 +68,12 @@ public Object fold() {
 
     @Override
     public int hashCode() {
-        return Objects.hash(value, dataType);
+        return Objects.hash(dataType, value);
```

Comment:
Changed the order of hashing to encourage caching / similarities of values that have the same datatype.

Reply:
Is there backwards compatibility concerns?